core/validation/schema_validation/validate.py

```diff
@@ -368,3 +370,36 @@ def validate_nullable(
                 )
 
     return non_nullable_constraint_failure
+
+
+def validate_project_dates(
+    data_dict: dict[str, pd.DataFrame], table: str, project_dates: list[str]
```

Comment:
`project_date_cols` or something similar may be a more accurate argument name here